addons/product/models/product_template.py

```diff
@@ -406,14 +406,23 @@ def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_ui
             return super(ProductTemplate, self)._name_search(name=name, args=args, operator=operator, limit=limit, name_get_uid=name_get_uid)
 
         Product = self.env['product.product']
+        Template = self.env['product.template']
         templates = self.browse([])
+        domain_no_variant = [('product_variant_ids', '=', False)]
+        domain_no_variant = domain_no_variant + args if args else domain_no_variant
         while True:
             domain = templates and [('product_tmpl_id', 'not in', templates.ids)] or []
             args = args if args is not None else []
             products_ns = Product._name_search(name, args+domain, operator=operator, name_get_uid=name_get_uid)
             products = Product.browse([x[0] for x in products_ns])
             templates |= products.mapped('product_tmpl_id')
-            if (not products) or (limit and (len(templates) > limit)):
+            this_round_templates = Template
```

Comment:
this_round_templates = self.browse([])

and `Template = self.env['product.template']` become useless

this_round_templates ?

Reply:
this_round as in "this round of loop"